src/lib/optimizer/abstract_syntax_tree/empty_node.cpp

```diff
@@ -0,0 +1,18 @@
+#include "empty_node.hpp"
+
+#include <memory>
+#include <string>
+
+#include "optimizer/abstract_syntax_tree/abstract_ast_node.hpp"
+
+namespace opossum {
+
+EmptyNode::EmptyNode() : AbstractASTNode(ASTNodeType::Empty) {}
+
+std::string EmptyNode::description() const { return "[EmptyNode]"; }
+
+std::shared_ptr<AbstractASTNode> EmptyNode::clone_subtree() const {
+  return _clone_without_subclass_members<decltype(*this)>();
```

Comment:
That's actually more typing than `_clone_without_subclass_members<EmptyNode>()`. I'd KISS this.

Reply:
The idea was to have the same code in all places. Unfortunately, there is no way to do this in the AbstractASTNode.

Reply:
With CRTP we could it in AbstractASTNode, and I actually like CRTP